### DISCUSSION TITLE
Support for multiple texture coordinates / UV maps

### DISCUSSION THREAD

**Discussion by jurisk** [bevyengine/bevy#8698] [Q&A] [answered]
When loading a glTF scene, I'm getting `WARN bevy_gltf::loader: Unknown vertex attribute TEXCOORD_1` errors, and the model loads, but without textures shown (it's completely grey).

I think this is because:
- The model has multiple UV Texture coordinate attributes defined
- https://github.com/bevyengine/bevy/blob/main/crates/bevy_gltf/src/vertex_attributes.rs#L251 only supports texture coordinate 0
- Likewise for https://github.com/bevyengine/bevy/blob/main/crates/bevy_render/src/mesh/mesh/mod.rs#L68

Two questions:
- Is there an open issue or RFC in Bevy for adding support for multiple sets of texture coordinates / UV maps?
- What are my options for loading / converting such a model?

**Comment by nicopap** [comment, marked as answer]
Yeah, you can open an issue as a feature request. #1802 lists "Multiple UVs" as a goal, but there isn't an open issue.

Looking at the GLtf spec (<https://registry.khronos.org/glTF/specs/2.0/glTF-2.0.html#meshes>), I see:

> Client implementations **SHOULD** support at least two texture coordinate sets, one vertex color, and one joints/weights set

searching for "texture coordinate" in the GLtf document, it seems multiple UVs are used to have a different set of coordinate per material textures (say the normal texture could map differently to the mesh than the diffuse/color texture) I don't quite see how we could implement it in bevy, but I think it's probably a worthwhile feature to consider.

**Comment by jurisk** [comment]
Thanks!

https://github.com/bevyengine/bevy/issues/8699 submitted.
